server/src/main/java/org/opensearch/OpenSearchException.java

```diff
@@ -1608,6 +1608,18 @@ private enum OpenSearchExceptionHandle {
             org.opensearch.index.shard.PrimaryShardClosedException::new,
             162,
             V_3_0_0
+        ),
+        DECOMMISSIONING_FAILED_EXCEPTION(
+            org.opensearch.cluster.decommission.DecommissioningFailedException.class,
+            org.opensearch.cluster.decommission.DecommissioningFailedException::new,
+            163,
+            V_3_0_0
```

Comment:
nit: DECOMMISSION_FAILED_EXCEPTION

Reply:
I kept it `DECOMMISSION_FAILED_EXCEPTION` but @shwetathareja suggested to change it to `DECOMMISSIONING_FAILED_EXCEPTION`